File: test_scripts/Real-Time Shipment Monitoring - Negative.js
describe('Real-Time Shipment Monitoring - Negative Test', () => {
  const loginPage = new LoginPage();
  const dashboardPage = new DashboardPage();
  const trackShipmentPage = new TrackShipmentPage();

  beforeEach(() => {
    cy.visit('/login');
  });

  it('should handle invalid shipment IDs gracefully', () => {
    // Step 1: Log in to the customer account using valid credentials
    loginPage.login('CUST98765', 'validPassword');
    dashboardPage.verifyLoginSuccess();

    // Step 2: Navigate to the 'Track Shipment' section
    dashboardPage.goToTrackShipment();
    trackShipmentPage.verifyTrackShipmentSectionDisplayed();

    // Step 3: Enter the Invalid Shipment ID in the search bar
    trackShipmentPage.enterShipmentId('SHIP00000');
    trackShipmentPage.verifyErrorMessageDisplayed('Invalid shipment ID');

    // Step 4: Attempt to refresh the page
    cy.reload();
    trackShipmentPage.verifyErrorMessageDisplayed('Invalid shipment ID');

    // Step 5: Log out of the customer account
    dashboardPage.logout();
    loginPage.verifyLogoutSuccess();
  });
});

File: step_definitions/Real-Time Shipment Monitoring - Negative.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import com.page_objects.ShipmentTrackingPage;

public class ShipmentTrackingSteps extends ShipmentTrackingPage {

    private TestHarness testHarness = new TestHarness();

    @Given("the user has a valid customer account with Customer ID {string}")
    public void theUserHasAValidCustomerAccountWithCustomerID(String customerId) {
        verifyCustomerAccount(customerId);
    }

    @Given("the user logs in using valid credentials")
    public void theUserLogsInUsingValidCredentials() {
        String username = testHarness.getData("LoginData", "UserName");
        String password = testHarness.getData("LoginData", "Password");
        login(username, password);
    }

    @Then("the user is successfully logged in to their account")
    public void theUserIsSuccessfullyLoggedInToTheirAccount() {
        verifyLoginSuccess();
    }

    @When("the user navigates to the 'Track Shipment' section")
    public void theUserNavigatesToTheTrackShipmentSection() {
        navigateToTrackShipmentSection();
    }

    @Then("the 'Track Shipment' section is displayed")
    public void theTrackShipmentSectionIsDisplayed() {
        verifyTrackShipmentSectionDisplayed();
    }

    @When("the user enters the Invalid Shipment ID {string} in the search bar")
    public void theUserEntersTheInvalidShipmentIDInTheSearchBar(String shipmentId) {
        enterShipmentID(shipmentId);
    }

    @Then("the system displays an error message indicating the shipment ID is invalid")
    public void theSystemDisplaysAnErrorMessageIndicatingTheShipmentIDIsInvalid() {
        verifyInvalidShipmentIDErrorMessage();
    }

    @When("the user attempts to refresh the page")
    public void theUserAttemptsToRefreshThePage() {
        refreshPage();
    }

    @Then("the error message persists, and no shipment details are displayed")
    public void theErrorMessagePersistsAndNoShipmentDetailsAreDisplayed() {
        verifyErrorMessagePersists();
    }

    @When("the user logs out of the customer account")
    public void theUserLogsOutOfTheCustomerAccount() {
        logout();
    }

    @Then("the user is successfully logged out")
    public void theUserIsSuccessfullyLoggedOut() {
        verifyLogoutSuccess();
    }
}